nav2_tasks/include/nav2_tasks/is_stuck_condition.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_TASKS__IS_STUCK_CONDITION_HPP
+#define NAV2_TASKS__IS_STUCK_CONDITION_HPP
+
+#include <string>
+#include <chrono>
+#include <ctime>
+#include <cmath>
+#include <thread>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/empty.hpp"
+#include "behavior_tree_core/condition_node.h"
+#include "nav2_robot/robot.hpp"
+
+namespace nav2_tasks
+{
+
+class IsStuckCondition : public BT::ConditionNode, public rclcpp::Node
+{
+
+public:
+  explicit IsStuckCondition(const std::string & condition_name)
+  : BT::ConditionNode(condition_name),
+    Node("IsStuckCondition")
+  {
+    RCLCPP_INFO(get_logger(), "IsStuckCondition::constructor");
+
+    vel_cmd_sub_ = this->create_subscription<geometry_msgs::msg::Twist>(
```

Comment:
These are both available via the robot class.

Reply:
The dwb controller is currently direclty publishing the velocity and need to change that so this condition node works correctly. Opened an issue #383.